src/pani/async.cljs

```diff
@@ -0,0 +1,85 @@
+(ns pani.async
+  (:require [pani.core :as p]
+            [cljs.core.async :refer [<! >! chan put! close!]])
+  (:require-macros [cljs.core.async.macros :refer [go go-loop]]))
+
+;; (defn chan-val [] (chan (map p/value)))
+
+;; (defn chan-snap [] (chan (map p/wrap-snapshot)))
+
+(defn with-chan "Dumb pseudo-macro" [f]
+  (let [ch (chan)]
+    (f ch)
+    ch))
+
+(defn chan->cb
+  "Create callback that pushes arguments onto given chan, and closes on nil"
+  [ch]
+  (fn [val]
+    (if val
+      (put! ch val)
+      (close! ch))))
+
+(defn chan->cb-once
+  "Create callback that pushes arguments onto chan exactly once"
+  [ch]
+  (fn [val]
+    (if val
+      (put! ch val))
```

Comment:
Not sure if that makes sense but `put!` has a third arity that could be used to close the channel on completion:
```clj
(put! ch val (fn [_] (close! ch)))
```